src/drivers/deploy/refresh.sh.template

```diff
@@ -20,21 +20,46 @@
 pushd $(dirname "$0") > /dev/null
 
 echo "Re-label all the node in the cluster!"
-echo "New machine with GPU label will be labeled on kubernetes"
+echo "Keep pods consistent with labels."
 
-/bin/bash node-label.sh || exit $?
-
-echo "Remove the label on the machine without GPU tag"
+# "For NotIn rule, it's for Daemonset, so just update label is enough, Daemonset's status will change with labels."
 
 {% for host in machinelist %}
+    {% for role in serviceinfo['deploy-rules'] %}
+        
+        {% if 'in' in role %}    
+            {% if role['in'] in machinelist[host] %}
+                kubectl label --overwrite=true nodes {{ machinelist[ host ][ 'nodename' ] }} {{role['in']}}='true' || exit $?
+                if ! `kubectl get po -o wide | grep -1 {{machinelist[ host ]['nodename']}} | grep -q {{servicename}}`; then
+                    echo "Start service {{servicename}} frome Node {{ machinelist[ host ][ 'nodename'] }} according to the cluster-configuration label definithion change"
+                    chmod +x {{serviceinfo['start-script']}}
+                    ./{{serviceinfo['start-script']}}
+                fi
+            {% else %}
+                if kubectl describe node {{ machinelist[ host ][ 'nodename' ] }} | grep -q {{role['in']}}='true'; then
+                    echo "Remove Node {{ machinelist[ host ][ 'nodename'] }}'s label, due to the cluster-configuration machinelist doesn't specify this label"
+                    kubectl label nodes {{ machinelist[ host ][ 'nodename' ] }} {{role['in']}}- || exit $?
+                fi
+                if `kubectl get po -o wide | grep -1 {{machinelist[ host ]['nodename']}} | grep -q {{servicename}}`; then
+                    echo "Stop service {{servicename}} frome Node {{ machinelist[ host ][ 'nodename'] }} according to the cluster-configuration label definithion change"
+                    chmod +x {{serviceinfo['stop-script']}}
+                    ./{{serviceinfo['stop-script']}}
```

Comment:
Little confused, if the label is removed, the pod should be stopped. Why we should running the stop script again? Attention, the stop script will stop all pod in the cluster.

Reply:
Only Daemonset will stopped with the label removed. The Deployment still running, need delete manually.